src/api/public/notes/notes.controller.ts

```diff
@@ -2,48 +2,51 @@ import {
   BadRequestException,
   Body,
   Controller,
+  createParamDecorator,
   Delete,
+  ExecutionContext,
   Get,
   Header,
   Logger,
   Param,
   Post,
   Put,
-  Req,
 } from '@nestjs/common';
-import { Request } from 'express';
 import * as getRawBody from 'raw-body';
 import { NotePermissionsUpdateDto } from '../../../notes/note-permissions.dto';
 import { NotesService } from '../../../notes/notes.service';
 import { RevisionsService } from '../../../revisions/revisions.service';
 
+/**
+ * Extract the raw markdown from the request body and create a new note with it
+ *
+ * Implementation inspired by https://stackoverflow.com/questions/52283713/how-do-i-pass-plain-text-as-my-request-body-using-nestjs
+ */
+export const PlainBody = createParamDecorator(async (_, context: ExecutionContext) => {
+  // we have to check req.readable because of raw-body issue #57
+  // https://github.com/stream-utils/raw-body/issues/57
+  const req = context.switchToHttp().getRequest<import('express').Request>();
+  if (req.readable) {
+    return (await getRawBody(req)).toString().trim();
+  }
+  throw new BadRequestException('Body is not text/plain');
```

Comment:
text/plain? I thought of text/markdown or is this a step forward for making the backend content-unaware.

Reply:
I've changed it to two more precise error messages.

Reply:
Now it's changed to text/markdown